File: resiliences-service/Payloads/CreateBeneficiaryRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using resiliences_service.Enums;

namespace resiliences_service.Payloads
{
    public class CreateBeneficiaryRequest
    {
        public uint UserId { get; set; }
        public BeneficiaryType BeneficiaryType { get; set; }
        public string BeneficiaryName { get; set; } = default!;
        public string Currency { get; set; } = "NGN";
        public string? AccountNumber { get; set; }
        public string? AccountName { get; set; }
        public string? BankCode { get; set; }
        public string? BankName { get; set; }
        public string? RecipientUsername { get; set; }
    }
}

File: epay/domain/src/main/java/com/epay/domain/beneficiary/input/CreateBeneficiaryRequest.java
package com.epay.domain.beneficiary.input;

import com.epay.domain.beneficiary.enums.BeneficiaryType;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;


@Data
public class CreateBeneficiaryRequest {

    @NotNull(message = "userId is required")
    private Long userId;

    @NotNull(message = "beneficiaryType is required")
    private BeneficiaryType beneficiaryType;

    @NotBlank(message = "beneficiaryName is required")
    private String beneficiaryName;

    private String currency = "NGN";

    private String accountNumber;
    private String accountName;
    private String bankCode;
    private String bankName;

    private String recipientUsername;
}
